### PR TITLE
Complete test environment for Toucan infra

### PR DESCRIPTION
**Pull request by kargakis**
Deploy the rest of the Toucan contracts necessary to have a TCO2 contract.

Plus cover all failure scenarios in unit tests.

### CODE REVIEW

**Comment by pheuberger** [comment]
can you add a description what is done here and why? :)

**Comment by kargakis** [comment]
@pheuberger description updated

File: lib/toucan.js [hunk 1]
```diff
@@ -10,20 +10,127 @@ async function deployProxy(contractName, registry, args = []) {
 	return contract;
 }
 
-async function prepareToucanEnv() {
```
Comment (pheuberger) [inline lib/toucan.js:13]: any reason why you're not using typescript?

Comment (pheuberger) [reply lib/toucan.js:13]: just curious, why did you prefix the function name with an underscore?

File: lib/toucan.js [hunk 2]
```diff
@@ -10,20 +10,127 @@ async function deployProxy(contractName, registry, args = []) {
...
+
+async function getProjectTokenId(tx) {
+	return _getNewTokenId(tx, "ProjectMinted");
+}
+
+async function getProjectVintageTokenId(tx) {
+	return _getNewTokenId(tx, "ProjectVintageMinted");
+}
+
+function yearStartTime(year) {
```
Comment (pheuberger) [inline lib/toucan.js:31]: does this return the unix timestamp of the first of january 0:00 in seconds?

File: lib/toucan.js (hunk 1 above)
Comment (kargakis) [reply lib/toucan.js:13]: > any reason why you're not using typescript?

No reason, can be refactored to TS as part of a separate ticket/PR.

> just curious, why did you prefix the function name with an underscore?

Just copy-pasta from the monorepo.

Comment (marceljay) [reply lib/toucan.js:13]: Which mono-repo? ^^

File: test/ToucanRegenBridge.js [hunk 3]
```diff
@@ -71,6 +71,12 @@ describe("Token contract", function () {
 				"Pausable: paused"
 			);
 		});
+
+		it("should fail with non-TCO2 contract", async function () {
+			await expect(bridge.connect(broker).bridge(recipient, tco2Factory.address, 10)).to.be.revertedWith(
```
Comment (marceljay) [inline test/ToucanRegenBridge.js]: Not an ERC20 too xD

**Review by marceljay — APPROVED** [review]
LGTM :)

**Pull request merged**